tests/MongoDB.Driver.Tests/Specifications/command-monitoring/CommandMonitoringTestRunner.cs

```diff
@@ -219,9 +234,40 @@ private void ExecuteOperation(IMongoDatabase database, IMongoCollection<BsonDocu
                 throw new SkipException(reason);
             }
 
+            var collection = GetCollection(database, collectionName, operation);
             test.Execute(__client.Cluster.Description, database, collection, arguments, async);
         }
 
+        private MongoCollectionSettings GetOperationCollectionOptions(BsonDocument collectionOptions)
```

Comment:
This is the opposite of the naming I was suggesting. For this method I was proposing:

    private MongoCollectionSettings ParseCollectionSettings(BsonDocument collectionSettings)
    {
        var settings = new MongoCollectionSettings();

        foreach (var collectionOption in collectionSettings.Elements)
        {
            switch (collectionOption.Name)
            {
                case "writeConcern":
                    settings.WriteConcern = WriteConcern.FromBsonDocument(collectionOption.Value.AsBsonDocument);
                    break;
                default:
                    throw new FormatException($"Unexpected collection option: {collectionOption.Name}.");
            }
        }

        return settings;
    }

Reply:
Sorry, my bad.
Fixed.